packages/dx-grid-core/src/plugins/sorting-state/reducers.js

```diff
@@ -1,39 +1,37 @@
-export const setColumnSorting = (sorting, {
-  columnName, direction, keepOther, cancel, scope,
+export const setColumnSorting = (state, {
+  columnName, direction, keepOther, cancel, sortIndex,
 }) => {
-  const sortingIndex = sorting.findIndex(s => s.columnName === columnName);
-  const columnSorting = sorting[sortingIndex];
-  let nextSorting = keepOther ? sorting.slice() : [];
+  const { sorting } = state;
 
-  if (scope) {
-    nextSorting = [
-      ...nextSorting,
-      ...sorting.filter(sort =>
-        scope.indexOf(sort.columnName) === -1 &&
-        sort.columnName !== columnName),
-    ];
+  let nextSorting = [];
+  if (keepOther === true) {
+    nextSorting = sorting.slice();
```

Comment:
Check case when seamless immutable is used. Probably, `sorting.slice()` should be replaced to `Array.from`.